src/pudl_archiver/archivers/nrel/nrelatb.py

```diff
@@ -77,76 +81,152 @@ def clean_excel_filename(
+                s3_viewer_url, s3_viewer_filename, zip_path
+            )
+            data_paths_in_archive.add(s3_viewer_filename)
+        return data_paths_in_archive
 
     async def get_year_type_resources(
         self,
         year: int,
         atb_type: Literal["transportation", "electricity"],
         year_excel_base_url: str,
     ):
-        """Get the transportation files."""
+        """Get the files for a year and type (electricity/transport)."""
         zip_path = self.download_directory / f"nrelatb-{year}-{atb_type}.zip"
         data_paths_in_archive = set()
 
-        year_parquet_url = f"{PARQUET_BASE_URL}{atb_type}%2Fparquet%2F{year}%2F"
-        parquet_urls = []
-        parquet_urls = await self.compile_parquet_urls(
-            parquet_urls, year_parquet_url, year, atb_type
-        )
-        for parquet_url in parquet_urls:
-            parquet_filename = (
-                parquet_url.split("/")[-1]
-                .lower()
-                .strip()
-                .replace(" ", "-")
-                .replace("_", "-")
+        # Compile URLs for Parquet and CSV files, download and add to archive
+
+        for file_type in ["csv", "parquet"]:
+            s3_viewer_urls = []
+            self.logger.info(
+                f"{year}/{atb_type}: Downloading {file_type} data from the S3 viewer."
             )
-            parquet_filename = f"nrelatb-{year}-{atb_type}-{parquet_filename}"
-            await self.download_add_to_archive_and_unlink(
-                parquet_url, parquet_filename, zip_path
+            year_file_url = f"{S3_VIEWER_BASE_URL}{atb_type}%2F{file_type}%2F{year}%2F"
+            s3_viewer_urls = await self.compile_s3_viewer_urls(
+                s3_viewer_urls=s3_viewer_urls,
+                url_to_check=year_file_url,
+                year=year,
+                atb_type=atb_type,
+                file_type=file_type,
+            )
+            data_paths_in_archive = await self.download_s3_viewer_urls(
+                s3_viewer_urls=s3_viewer_urls,
+                data_paths_in_archive=data_paths_in_archive,
+                zip_path=zip_path,
+                year=year,
+                atb_type=atb_type,
+                file_type=file_type,
             )
-            data_paths_in_archive.add(parquet_filename)
 
         # now for the excel/csv files
         link_pattern = re.compile(r"(.*)(.xlsx|.xlsm|.csv|.zip)$")
+        self.logger.info(
+            f"{year}/{atb_type}: Downloading spreadsheet data from {year_excel_base_url}."
+        )
         for excel_url in await self.get_hyperlinks(year_excel_base_url, link_pattern):
             excel_filename = self.clean_excel_filename(excel_url, year, atb_type)
-            excel_url = urljoin(year_excel_base_url, excel_url)
-            await self.download_add_to_archive_and_unlink(
-                excel_url, excel_filename, zip_path
+            if excel_filename not in data_paths_in_archive:  # Avoid duplicates
```

Comment:
why does this avoid dupes need to happen? bc the s3 compile/download step grabs CSV's? should we take out the .csv in the link pattern instead?